From 0c16775b5cedca6205a28148fa6f2a111ae499c6 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Tue, 26 Jul 2022 08:45:02 -0300
Subject: [PATCH 053/142] i386: Disable math_opt_barrier and math_force_eval
 for clang

It triggers an ICE with clang while building round_and_return.
---
 sysdeps/x86/fpu/math-barriers.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysdeps/x86/fpu/math-barriers.h b/sysdeps/x86/fpu/math-barriers.h
index 6fdcdac280..59577b77ea 100644
--- a/sysdeps/x86/fpu/math-barriers.h
+++ b/sysdeps/x86/fpu/math-barriers.h
@@ -19,7 +19,7 @@
 #ifndef X86_MATH_BARRIERS_H
 #define X86_MATH_BARRIERS_H 1
 
-#ifdef __SSE2_MATH__
+#if defined __SSE2_MATH__ && !defined (__clang__)
 # define math_opt_barrier(x)						\
   ({ __typeof(x) __x;							\
      if (sizeof (x) <= sizeof (double)					\
-- 
2.34.1

